python/setup.py

```
@@ -63,107 +141,273 @@ def _find_eigen_include_dir():
-            language="c++",
-            extra_compile_args=(
+    Real LIB/ sources (monorepo ``../LIB`` or sdist ``LIB/``) are attached here
+    so ``setup()`` only ever sees in-tree relative paths.
+    """
+
+    def build_extensions(self) -> None:
+        if not self.extensions:
+            return
+        try:
+            super().build_extensions()
+        except Exception as exc:  # pragma: no cover - platform/compiler dependent
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)